cli/commands/asset/create.go

```diff
@@ -32,7 +28,14 @@ func CreateCommand(cli *cli.SensuCli) *cobra.Command {
 				_ = cmd.MarkFlagRequired("url")
 			}
 		},
-		RunE: exec.Run,
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Out of curiosity, what's the difference here between these two formats?

Reply:
I'm assuming you're asking about the difference here between assigning `exec.Run` vs assigning an anonymous function. No difference!

The important change here is that I moved the creation of `executor` and the namespace it holds from command creation time to command run time.

I made that change before converging on my actual, final solution of using the parse error whitelisting facilities of newer versions of Cobra. I suspect that this specific change in `asset/create.go` might be unnecessary now: since the commands are created after the flags are parsed, `cli.Config.Namespace()` will probably have the right value.

That being said, I think keeping this change makes sense because it makes it so the command uses the namespace that's defined when it _runs_ rather than when it's _defined_. An important difference, especially if there is a refactor that creates all the commands before parsing the flags!